workflow-examples/src/main/java/com/redhat/parodos/examples/move2kube/task/Move2KubePlan.java

```diff
@@ -0,0 +1,93 @@
+package com.redhat.parodos.examples.move2kube.task;
+
+import java.io.File;
+import java.util.List;
+
+import com.redhat.parodos.workflow.task.enums.WorkFlowTaskOutput;
+import com.redhat.parodos.workflows.work.DefaultWorkReport;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import dev.parodos.move2kube.ApiClient;
+import dev.parodos.move2kube.ApiException;
+import dev.parodos.move2kube.api.PlanApi;
+import dev.parodos.move2kube.api.ProjectInputsApi;
+import dev.parodos.move2kube.client.model.GetPlan200Response;
+import lombok.extern.slf4j.Slf4j;
+
+import static java.lang.Thread.sleep;
+
+@Slf4j
+public class Move2KubePlan extends Move2KubeBase {
+
+	PlanApi planApi;
+
+	ProjectInputsApi projectInputsApi;
+
+	public Move2KubePlan(String server) {
+		super();
+		this.setClient(server);
+		planApi = new PlanApi(client);
+
+		ApiClient clientFormData = client;
+		clientFormData.addDefaultHeader("Content-Type", "multipart/form-data");
+		ProjectInputsApi projectInputsApi = new ProjectInputsApi(clientFormData);
+	}
+
+	// only used for testing
+	public Move2KubePlan(String server, PlanApi plan, ProjectInputsApi projectInputs) {
+		super();
+		this.setClient(server);
+		planApi = plan;
+		projectInputsApi = projectInputs;
+	}
+
+	public WorkReport execute(WorkContext workContext) {
+		// PlanApi planApi = new PlanApi(client);
+		String workspaceID = (String) workContext.get(getWorkspaceContextKey());
+		String projectID = (String) workContext.get(getProjectContextKey());
+		addSourceCode(workspaceID, projectID, (String) workContext.get("gitArchivePath"));
+		try {
+			planApi.startPlanning(workspaceID, projectID);
+
+			for (int i = 1; i <= 10; ++i) {
+				GetPlan200Response plan = planApi.getPlan(workspaceID, projectID);
```

Comment:
Why do we want to wait exponentially here to get the plan? We could try using [async](https://github.com/eloycoto/move2kube-java-client/blob/main/java-client/src/main/java/dev/parodos/move2kube/api/PlanApi.java#L363) instead. Please take a look how @gciavarrini [implemented](https://github.com/parodos-dev/parodos/blob/6af518ef87b9b922720bc15135dcc3855daf9df0/sdk-utils/src/main/java/com/redhat/parodos/sdkutils/SdkUtils.java#L114) api wait.

Reply:
That was exactly the reason why Gloria implemented it in this way.

Reply:
ohh, got it. The main problem that I can see, I need to implement the async way for move2kube API for an example workflow, does that make sense?